### PR TITLE
Add stubs for transaction native functions

### PR DESCRIPTION
**Pull request by alnoki**
## Motivation

(Write your motivation for proposed changes here.)

The proposed native functions enable the functionality described during a previous call with core Aptos developers, namely a parallelized matching engine for a central limit order book. The code changes in this pull request are only stubs, designed to initiate a conversation about future development.

The proposed native function `AptosFramework::Transaction::version_number` returns the Aptos ledger version number during transaction execution, which allows for chronological re-sequencing of data structures within Move, when they have been written to non-overlapping locations in memory. Through Block-STM's optimistic concurrency, non-overlapping write sets execute in parallel, and if the data structure in each write set can be marked with the version number as proposed through this native function, then it is possible for a matching engine to determine chronology for data structures that were written, for example, to queues that were designed to take in orders in parallel.

The second proposed native function `AptosFramework::Transaction::init_schedule` initializes a regular schedule for transaction execution of a public script function, authorized by the signing account. This is basically a transaction scheduler that enables a matching engine on a central limit order book to be invoked on a regular interval, for example through a simple trigger function of the form `Orderbook::Match::match`. Without a transaction scheduler, the matching engine would require external inputs to initiate dispersement of funds, analagous to "crank turners" on the Serum DEX from Solana.

The third proposed native function, `AptosFramework::Transaction::cancel_schedule` ends the recurring schedule for a public script function that had been initialized per above.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes, I have, and I have signed a CLA

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/910)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/Transaction.move [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    );
+
+    /// Cancel the recurring schedule for a transaction previously
+    /// authorized by `account`
+    public(script) native fun cancel_schedule(
+        account: &signer,
+        script_address: address,
+        script_module_name: String,
+        script_function_name: String,
+    );
```
Comment (msmouse) [inline aptos-move/framework/aptos-framework/sources/Transaction.move:31]: I'd remove this and let the "hook function" reschedule itself every time.

File: aptos-move/framework/aptos-framework/sources/Transaction.move [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public(script) native fun schedule(
+        account: &signer,
+        script_address: address,
+        script_module_name: String,
+        script_function_name: String,
+        type_arguments: &vector<TypeInfo>,
+        arguments: &vector<String>,
+        delay: u64,
+    );
+}
```
Comment (alnoki) [inline aptos-move/framework/aptos-framework/sources/Transaction.move:28]: @msmouse How about this? Based on past discussion with the core dev team about an event scheduler, I'm assuming that scheduled transactions would be executed during what I'm calling the "block epilogue"

Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/Transaction.move:28]: This seems overall okay as a concept, do you want to build out the idea?

Comment (alnoki) [reply aptos-move/framework/aptos-framework/sources/Transaction.move:28]: @davidiw Sounds like a plan. Right now the order book's atomic matching engine is being implemented, and when that is complete next up is parallelized order queueing (which leverages the version number native function in this proposal), and automated batchwise order matching on a regular interval (which leverages the event scheduler in this proposal). It's looking like the proposed functionality may require integration with the [parallel executor scheduler](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/parallel-executor/src/executor.rs) and the Move VM, and I look forward to discussing this further when the incubation program starts and in-person collaboration is possible

**Comment by davidiw** [comment]
Gonna close this out for now.

**Comment by davidiw** [comment]
Let's either move this over to an issue or a google doc!

**Comment by alnoki** [comment]
> 

Split into #1975 and #1977, see below

**Pull request closed without merging**